driver/src/test/scala/UpdateSpec.scala

```diff
@@ -62,6 +59,55 @@ trait UpdateSpec extends UpdateFixtures { collectionSpec: CollectionSpec =>
       section("gt_mongo32")
     }
 
+    "update many documents" >> {
+      case class TestDocument(id: String, children: List[BSONDocument])
+
+      def mockChildElements =
+        (childrenNumber: Int) =>
+          List
+            .range(0, childrenNumber)
+            .map(
+              num =>
+                BSONDocument(
+                  "id" -> BSONString(s"child${num}"),
+                  "value" -> BSONString("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa")))
+
+      // Method Under Test
+      def upsertElements(collection: DefaultCollection, elements: List[TestDocument]) = {
+        val updateBuilder = collection.update(ordered = false)
+        val updateElements = Future.sequence(
+          for {
+            element <- elements
+          } yield updateBuilder.element(
+            q = BSONDocument("id" -> element.id),
+            u = BSONDocument(
+              "id" -> element.id,
+              "children" -> BSONArray(element.children)),
+            upsert = true))
+
+        updateElements.flatMap(updates => updateBuilder.many(updates))
+      }
+
+      def spec(c: DefaultCollection, timeout: FiniteDuration) = {
+        val childrenPerElement = 200 * 1000
+        val elements = List(
+          TestDocument("id1", mockChildElements(childrenPerElement)),
+          TestDocument("id2", mockChildElements(childrenPerElement)))
+
+        val writeResult = Await.result(upsertElements(c, elements), timeout)
+        // Wait to make sure the operation is fully finished.
+        Thread.sleep(1000L)
```

Comment:
This is not really needed. Added only for a demonstration.

Reply:
So remove it

Reply:
Done.